herckules/src/main/java/edu/asu/diging/lerna/herckules/db/impl/ProjectManager.java

```diff
@@ -0,0 +1,118 @@
+package edu.asu.diging.lerna.herckules.db.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import edu.asu.diging.lerna.herckules.authentication.impl.IExceptionMessages;
+import edu.asu.diging.lerna.herckules.db.IProjectDBConnector;
+import edu.asu.diging.lerna.herckules.db.IProjectManager;
+import edu.asu.diging.lerna.herckules.domain.impl.Project;
+import edu.asu.diging.lerna.herkules.exception.HerckulesArgumentException;
+import edu.asu.diging.lerna.herkules.exception.ProjectIDNotFoundException;
+
+/**
+ * This class implements {@link IProjectManager} functions. This manager class
+ * connects the project controller and the ProjectDBConnector. This is used to
+ * validate business logic before retrieving, adding, updating, or deleting a
+ * project.
+ * 
+ * @author Tamalika Mukherjee
+ * @author Vineel Vutukuri
+ */
+
+@Service
+public class ProjectManager implements IProjectManager {
+	@Autowired
+	private IProjectDBConnector dbconnector;
+
+	/**
+	 * Deletes the project from the database after validation of parameter.
+	 * 
+	 * @param projectid
+	 *            The identifier for the project to be deleted.
+	 * @return true if the project is deleted from the database.
+	 * @throws ProjectIDNotFoundException
+	 *             Exception thrown when project with the specified ID does not
+	 *             exist.
+	 * @throws HerckulesArgumentException
+	 *             Exception thrown if the provided projectID is empty or null
+	 */
+
+	public boolean deleteProject(String projectid)
+			throws HerckulesArgumentException, ProjectIDNotFoundException {
+
+		if (projectid == null || projectid.trim().equals("")) {
+			throw new HerckulesArgumentException(
+					IExceptionMessages.EMPTY_INPUT_EXCEPTION);
+
+		}
+		return dbconnector.deleteProject(projectid);
+	}
+
+	/**
+	 * Retrieves the project from the database after validation of parameter.
+	 * 
+	 * @param project
+	 *            The project object that is to be retrieved.
+	 * @return true if the project is retrieved from the database.
+	 * @throws HerckulesArgumentException
+	 *             Exception thrown if the provided projectID is empty or null
+	 */
+	public Project retrieveProject(Project project)

```

Comment:
This doesn't make too much sense. You probably want to retrieve a project by its id, and/or creator. But why would you give a project to retrieve the same?